client/src/client.rs

```diff
@@ -235,13 +235,33 @@ impl<B, E, Block, RA> Client<B, E, Block, RA> where
 	}
 
 	/// Given a `BlockId` and a key prefix, return the matching child storage keys in that block.
-	pub fn storage_keys(&self, id: &BlockId<Block>, key_prefix: &StorageKey) -> sp_blockchain::Result<Vec<StorageKey>> {
+	pub fn storage_keys(&self, id: &BlockId<Block>, key_prefix: &StorageKey)
+		-> sp_blockchain::Result<Vec<StorageKey>>
+	{
 		let keys = self.state_at(id)?.keys(&key_prefix.0).into_iter().map(StorageKey).collect();
 		Ok(keys)
 	}
 
+	/// Given a `BlockId` and a key prefix, return the matching child storage keys and values in that block.
+	pub fn storage_pairs(&self, id: &BlockId<Block>, key_prefix: &StorageKey)
+		-> sp_blockchain::Result<Vec<(StorageKey, StorageData)>>
+	{
+		let state = self.state_at(id)?;
+		let keys = state
+			.keys(&key_prefix.0)
+			.into_iter()
+			.map(|k| {
+				let d = state.storage(&k).unwrap_or(None).unwrap_or(vec![]);
```

Comment:
```suggestion
				let d = state.storage(&k).ok().unwrap_or_default();
```

Reply:
that doesn't do the same thing, does it?

Reply:
Yeah, I had forgotten a `.flatten()` :( Sorry. FWIW, I also merged master now.